src/components/Rating/Rating.tsx

```diff
@@ -0,0 +1,124 @@
+import { useEffect, useState } from "react";
+import "./rating.scss"; // Assuming you have a Sass file for styling
+
+
+export interface RatingProps {
+    totalStars: number;
+    shape?: string;
+    disabled?: boolean
+    readOnly?: boolean;
+    readOnlyValue?: number
+}
+
+const Rating: React.FC<RatingProps> = ({
+    totalStars = 5,
+    shape,
+    disabled = false,
+    readOnly,
+    readOnlyValue = totalStars
+}) => {
+    const numberOfStars = totalStars
+    const [rating, setRating] = useState(0);
+    const [hoveredRating, setHoveredRating] = useState(0);
+
+
+    useEffect(() => {
+        if (readOnly) {
+            setRating(readOnlyValue)
```

Comment:
this should be derived instead of set, which you can do by creating a conditional derived value ahead of useState.

Never _set_ where you can _derive_

```tsx
const defaultRating = readOnly ? readOnlyValue : 0;
const [rating, setRating] = useState(default);

```